AnkiDroid/src/main/java/com/ichi2/anki/pages/AnkiPagesWebview.kt

```diff
@@ -0,0 +1,86 @@
+ * this program.  If not, see <http://www.gnu.org/licenses/>.                           *
+ ****************************************************************************************/
+
+package com.ichi2.anki.pages
+
+import android.os.Bundle
+import android.webkit.*
+import com.ichi2.anki.AnkiActivity
+import com.ichi2.anki.R
+import com.ichi2.libanki.Collection
+import timber.log.Timber
+import java.io.ByteArrayInputStream
+import java.io.InputStream
+
+class AnkiPagesWebview : AnkiActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.anki_pages_webview)
+        val webview: WebView = findViewById(R.id.anki_wb)
+        webview.settings.javaScriptEnabled = true
+        webview.settings.allowFileAccess = true
+        webview.webChromeClient = WebChromeClient()
+        // passing col in here is presumably the wrong way to do this
+        webview.webViewClient = AnkiWebChromeClient(col)
+        webview.loadUrl("https://127.0.0.1/graphs.html")
+    }
+
+    class AnkiWebChromeClient(val col: Collection) : WebViewClient() {
+        override fun shouldInterceptRequest(
+            view: WebView?,
+            request: WebResourceRequest?
+        ): WebResourceResponse? {
+            val streamResponse = { data: InputStream, mime: String ->
+                WebResourceResponse(
+                    mime, "utf-8", 200, "OK",
+                    HashMap(), data
+                )
+            }
+
+            val fileResponse = { path: String, mime: String ->
+                val inputStream: InputStream = this.javaClass.classLoader!!.getResourceAsStream("web$path")
+                streamResponse(inputStream, mime)
+            }
+
+            return request?.url?.let { url ->
+                Timber.i("************** request %s", url)
+                when (val path = url.path) {
+                    "/graphs.html" -> fileResponse(path, "text/html")
+                    "/graphs.css" -> fileResponse(path, "text/css")
+                    "/graphs-base.css" -> fileResponse(path, "text/css")
+                    "/graphs.js" -> fileResponse(path, "text/javascript")
+                    "/_anki/i18nResources" -> {
+                        val byteArray: ByteArray = "en-US".toByteArray()
+                        val data = ByteArrayInputStream(col.backend.i18nResourcesRaw(byteArray))
+                        streamResponse(data, "application/binary")
+                    }
+                    "/_anki/getGraphPreferences" -> {
+                        val data = ByteArrayInputStream(col.backend.getGraphPreferences().toByteArray())
+                        streamResponse(data, "application/binary")
+                    }
+                    "/_anki/graphs" -> {
+                        val data = ByteArrayInputStream(col.backend.graphs("deck:current", 365).toByteArray())
```

Comment:
You should be able to use graphsRaw() now, and pass it the bytes from the request body.

Reply:
Thanks, I am implementing it.